operator/controller/src/test/java/io/apicurio/registry/operator/it/TlsITTest.java

```diff
@@ -0,0 +1,162 @@
+package io.apicurio.registry.operator.it;
+
+import io.apicurio.registry.operator.EnvironmentVariables;
+import io.apicurio.registry.operator.api.v1.ApicurioRegistry3;
+import io.apicurio.registry.operator.resource.ResourceFactory;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.networking.v1.NetworkPolicyIngressRule;
+import io.fabric8.kubernetes.client.utils.Serialization;
+import io.quarkus.test.junit.QuarkusTest;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static io.apicurio.registry.operator.api.v1.ContainerNames.REGISTRY_APP_CONTAINER_NAME;
+import static io.apicurio.registry.operator.resource.app.AppDeploymentResource.getContainerFromDeployment;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+@QuarkusTest
+public class TlsITTest extends ITBase {
```

Comment:
We should ping the REST API in the tests, port-forwarding should be enough since we do not support yet create HTTPS Ingresses.

Reply:
If https is not properly configured at the application level, the pod won't even start, so I think checking whether the deployment is ready or not should be enough.

Reply:
See above, the application might be configured correctly but the connection still could not be made. I'm suggesting to test that the deployment is correct from the cluster PoV.